xformers/factory/block_factory.py

```diff
@@ -175,10 +175,7 @@ def __init__(self, config: xFormerEncoderConfig, **kwargs):
         # Optional patch embedding
         self.patch_emb: Optional[nn.Module] = None
 
-        if (
```

Comment:
this was not respecting the config, while trying to do the right thing: if the config was asking for a patch embedding, but the layer was not first, it would not be instantiated. In retrospect I think that it's risky, not doing what the API says it will do, plus it only worked in practice because is_first() was often wrong. I think now that it's better to respect the config no matter what, and not silently diverge